pkg/server/config.go

```diff
@@ -0,0 +1,494 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"reflect"
+	"time"
+
+	configroot "github.com/kcp-dev/kcp/config/root"
+	configrootphase0 "github.com/kcp-dev/kcp/config/root-phase0"
+	systemcrds "github.com/kcp-dev/kcp/config/system-crds"
+	kcpadmissioninitializers "github.com/kcp-dev/kcp/pkg/admission/initializers"
+	apisv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+	tenancyv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/tenancy/v1alpha1"
+	bootstrappolicy "github.com/kcp-dev/kcp/pkg/authorization/bootstrap"
+	kcpclient "github.com/kcp-dev/kcp/pkg/client/clientset/versioned"
+	kcpexternalversions "github.com/kcp-dev/kcp/pkg/client/informers/externalversions"
+	"github.com/kcp-dev/kcp/pkg/etcd"
+	"github.com/kcp-dev/kcp/pkg/indexers"
+	"github.com/kcp-dev/kcp/pkg/informer"
+	metadataclient "github.com/kcp-dev/kcp/pkg/metadata"
+	boostrap "github.com/kcp-dev/kcp/pkg/server/bootstrap"
+	"github.com/kcp-dev/kcp/pkg/server/options"
+	"github.com/kcp-dev/kcp/pkg/server/requestinfo"
+	"github.com/kcp-dev/logicalcluster"
+	etcdtypes "go.etcd.io/etcd/client/pkg/v3/types"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	apiextensionsexternalversions "k8s.io/apiextensions-apiserver/pkg/client/informers/externalversions"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/apiserver/pkg/endpoints/filters"
+	genericapiserver "k8s.io/apiserver/pkg/server"
+	"k8s.io/apiserver/pkg/util/webhook"
+	"k8s.io/client-go/dynamic"
+	coreexternalversions "k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/client-go/tools/clusters"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/genericcontrolplane"
+)
+
+type Config struct {
+	options *options.CompletedOptions
+}
+
+type CompletedConfig struct {
+	EtcdServer                            *etcd.Server
+	kcpSharedInformerFactory              kcpexternalversions.SharedInformerFactory
+	kubeSharedInformerFactory             coreexternalversions.SharedInformerFactory
+	apiextensionsSharedInformerFactory    apiextensionsexternalversions.SharedInformerFactory
+	dynamicDiscoverySharedInformerFactory *informer.DynamicDiscoverySharedInformerFactory
+
+	identityConfig        *rest.Config
+	serverChain           *genericcontrolplane.ServerChain
+	server                *genericapiserver.GenericAPIServer
```

Comment:
this is not config level, but server. See my other comment that this struct is actually what Server should be. You lack the config level.